core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -176,7 +192,17 @@ public void validateIndexName(String index, ClusterState state) {
         }
     }
 
-    public void createIndex(final CreateIndexClusterStateUpdateRequest request, final ActionListener<ClusterStateUpdateResponse> listener) {
+    public void createIndex(final CreateIndexClusterStateUpdateRequest request,
+                            final ActionListener<ClusterStateUpdateResponse> listener) {
+
+        final int numShardCopies = IndexMetaData.INDEX_NUMBER_OF_REPLICAS_SETTING.get(request.settings()) + 1;
+        if (request.waitForActiveShards().exceeds(numShardCopies)) {
```

Comment:
Maybe it's the simplest to add the waiting to the TransportCreateIndexAction ? is there any good reason to inline it here?